data/common/src/main/java/com/example/data/common/api/di/NetworkModule.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.data.common.api.di
+
+import com.example.data.common.BuildConfig
+import com.example.data.common.api.CoinCapRetrofit
+import com.example.data.common.api.RetrofitBuilder
+import com.jakewharton.retrofit2.adapter.kotlin.coroutines.CoroutineCallAdapterFactory
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+/**
+ * @author yaya (@yahyalmh)
+ * @since 04th November 2022
+ */
+
+@Module
+@InstallIn(SingletonComponent::class)
+interface NetworkModule {
+    @Binds
+    fun bindCoincapRetrofit(coincapRetrofit: CoinCapRetrofit): RetrofitBuilder
+
+    companion object {
+        @Provides
+        fun provideHttpClient(): OkHttpClient = if (BuildConfig.DEBUG) {
+            val logging = HttpLoggingInterceptor()
+            logging.setLevel(HttpLoggingInterceptor.Level.BODY)
+            OkHttpClient.Builder()
+                .readTimeout(60, TimeUnit.SECONDS)
+                .writeTimeout(60, TimeUnit.SECONDS)
+                .addInterceptor(logging)
```

Comment:
Learning - read a little bit about interceptors. You can do a lot of useful stuff with custom interceptors :)

Reply:
I have a question maybe related to interceptors. envisage, an api return different response type for the same request, how can we handle it with retrofit?
The ok response when I have permission:
`{
method: {
_content: "flickr.test.echo"
},
format: {
_content: "json"
},
api_key: {
_content: "8cb5d1fd54f34b91543e5968a1ae9cde"
},
stat: "ok"
}`

The same request return fail response, when I do not have permission:

`{
stat: "fail",
code: 99,
message: "Insufficient permissions. Method requires read privileges; none granted."
}`

Reply:
In short: Values returned by retrofit should be wrapped. One of wrappers is Response class. It contain method isSuccessful. So whenever you get a response you can do something like:
`If (response.isSuccessful) { do something based on response.body value }  else { display error message to the user, unlock the view if you are using custom spinner/view locker }`
Write me on teams if you want to talk this through

Reply:
Here is some short example - someone is trying to get QuoteList via retrofit. At the end of tutorial there is response of type Response<QuoteList>
https://www.geeksforgeeks.org/retrofit-with-kotlin-coroutine-in-android/